### PR TITLE
code snippet to find “  The maximum inner circle of the contour”

### PR DESCRIPTION
**Pull request by jsxyhelu**
add some code to find the maximum enclosed circle by using pointPolygonTest
I think this is the right place to pull requests
hope pass

```
force_builders=linux,Docs
```

### CODE REVIEW

**Comment by jsxyhelu** [comment]
default

**Comment by alalek** [comment]
Could you please add result image into corresponding tutorial? (you can save them via `imwrite()` from sample - please use jpg format)

This patch should [go](https://github.com/opencv/opencv/wiki/How_to_contribute#faq) into 3.4 branch. We will merge changes from 3.4 into master regularly (weekly/bi-weekly).

So, please:
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

File: samples/cpp/tutorial_code/ShapeDescriptors/pointPolygonTest_demo.cpp [hunk 1]
```diff
@@ -77,9 +92,28 @@ int main( void )
...
 
+	//get the biggest Contour
+	vector<Point> biggestContour = FindBiggestContour(src);
+	//find the maximum enclosed circle 
+	int dist = 0;
+	int maxdist = 0;
+	Point center;
+	for(int i=0;i<src.cols;i++)
+	{
+		for(int j=0;j<src.rows;j++)
```
[the remark this answers is not in the document]
Comment (jsxyhelu) [reply samples/cpp/tutorial_code/ShapeDescriptors/pointPolygonTest_demo.cpp]: yes,you are right

**Comment by jsxyhelu** [comment]
have changed the base branch from master to 3.4 
wait to pass

**Comment by jsxyhelu** [comment]
distance_and_maximum_enclosed_circle

**Comment by alalek** [comment]
Please perform step 2 and 3 from "rebasing" comment above.

**Comment by jsxyhelu** [comment]
alalek ,in my machine pointPolygonTest can can run from beginning to end，sorry for my poor english,I did not get your idea by committing 
> Please perform steps 2 and 3 from "rebasing" comment above.
by the way ,I have pull "The_maximum_inner_circle_result.jpg " to "opencv/doc/tutorials/imgproc/shapedescriptors/point_polygon_test/images/" ，I believe it is the right place to save it.
I hope  very much that my first “pull request ” to opencv can be passed.And I think I did submit something useful.
Wait on line……

**Comment by jsxyhelu** [comment]
Wait on line

**Comment by jsxyhelu** [comment]
opencv do not have a function to find “ The maximum inner circle of the contour”,
but we can do this work by using pointPolygonTest
code snippet and Tutorial picture about it,hope pass

**Comment by alalek** [comment]
I rebased this patch to 3.4 branch. Update code from your fork on your side.

**Comment by jsxyhelu** [comment]
code  updated.

**Comment by alalek** [comment]
> Merge pull request 1 from opencv/3.4  

This is wrong. Also it is not required at all.
All patched should be prepared without any "merge" commits.

Will reset it back again.

**Comment by jsxyhelu** [comment]
alalek 
sorry for my lack of experience，but I really don't known what to do next?
hope to be able to take you a little time, tell me how to make it pass?
thanks ,from the bottom of my heart!

**Comment by jsxyhelu** [comment]
have Update my fork  to 3.4 branch

**Comment by jsxyhelu** [comment]
close this pull request
i will read "How_to_contribute" again,and do it again.

**Pull request closed without merging**